dlp/deIdentifyWithRedaction.js

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//  title: De-identify data: Redacting with matched input values
+//  description: Uses the Data Loss Prevention API to de-identify sensitive data
+//    in a string by redacting matched input values.
+//  usage: node deIdentifyWithRedaction.js my-project textToInspect infoTypes
+
+function main(projectId, textToInspect, infoTypes) {
+  infoTypes = transformCLI(infoTypes);
+  // [START dlp_deidentify_redact]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient();
+
+  // TODO(developer): Replace these variables before running the sample.
+  // const projectId = "your-project-id";
+
+  // The string to deidentify
+  // const textToInspect =
+  //   'My name is Alicia Abernathy, and my email address is aabernathy@example.com.';
+
+  // The infoTypes of information to match
+  // See https://cloud.google.com/dlp/docs/concepts-infotypes for more information
+  // about supported infoTypes.
+  // const infoTypes = [{name: 'EMAIL_ADDRESS'}];
+
+  async function deIdentifyRedaction() {
+    // Construct deidentify configuration
+    const deidentifyConfig = {
+      infoTypeTransformations: {
+        transformations: [
+          {
+            primitiveTransformation: {
+              redactConfig: {},
+            },
+          },
+        ],
+      },
+    };
```

Comment:
The JSON equivalent recommends setting infoTypes in the infoTypeTranformations.transformations. I see that is missing here. 
```
    "infoTypeTransformations":{
      "transformations":[
        {
          "infoTypes":[
            {
              "name":"EMAIL_ADDRESS"
            }
          ],
          "primitiveTransformation":{
            "redactConfig":{

            }
          }
        }
      ]
    }
  }
```